src/track/heaptrack_inject.cpp

```diff
@@ -176,9 +176,80 @@ struct dlopen
     static constexpr auto name = "dlopen";
     static constexpr auto original = &::dlopen;
 
+    static Dl_serinfo* load_serinfo(const char* module_path) noexcept
```

Comment:
instead of returning an memory allocated serinfo, can't you take that `Dl_serinfo *` as an out-param, and construct it on the stack of the caller and return bool instead?